code/game/machinery/computer/buildandrepair.dm

```diff
@@ -258,26 +257,24 @@
 	origin_tech = "programming=2"
 
 
-/obj/item/weapon/circuitboard/supplycomp/attackby(obj/item/I as obj, mob/user as mob)
-	if(istype(I,/obj/item/device/multitool))
-		var/catastasis = src.contraband_enabled
-		var/opposite_catastasis
-		if(catastasis)
-			opposite_catastasis = "STANDARD"
-			catastasis = "BROAD"
-		else
-			opposite_catastasis = "BROAD"
-			catastasis = "STANDARD"
-
-		switch( alert("Current receiver spectrum is set to: [catastasis]","Multitool-Circuitboard interface","Switch to [opposite_catastasis]","Cancel") )
-		//switch( alert("Current receiver spectrum is set to: " {(src.contraband_enabled) ? ("BROAD") : ("STANDARD")} , "Multitool-Circuitboard interface" , "Switch to " {(src.contraband_enabled) ? ("STANDARD") : ("BROAD")}, "Cancel") )
-			if("Switch to STANDARD","Switch to BROAD")
-				src.contraband_enabled = !src.contraband_enabled
-
-			if("Cancel")
+/obj/item/weapon/circuitboard/attackby(obj/item/I as obj, mob/user as mob)
+	if(issolder(I))
+		var/obj/item/weapon/solder/S = I
+		if(S.remove_fuel(2,user))
+			if(istype(src,/obj/item/weapon/circuitboard/supplycomp))

```

Comment:
Rather than typecasting this (you should really never cast src, it's a worrying thing) I suggest making a solder_act() proc of some kind.